agroal-api/src/main/java/io/agroal/api/configuration/AgroalConnectionPoolConfiguration.java

```diff
@@ -93,6 +93,30 @@ public interface AgroalConnectionPoolConfiguration {
      */
     boolean flushOnClose();
 
+    /**
+     * If connections from this pool should be used during recovery. For XA pools it is vital that recovery is enabled
+     * on at least one connection pool to the same database and therefore the default is true.
+     * <p>
+     * Normally a transaction manager will call xa_recover () on a connection during recovery to obtain a list of
+     * transaction branches that are currently in a prepared or heuristically completed state. However, it can
+     * happen that multiple XA connections connect to the same database which would all return the same set of branches
+     * and for performance reasons only one should be used for recover() calls.
+     * <p>
+     * @return true if enabled
+     */
+    default boolean recoveryEnable() {
+        return true;
+    }
+
+    /**
+     * Whether recover() calls should be used on this pool during recovery.
+     *
+     * @param enable
+     */
+    default void setRecoveryEnable(boolean enable) {
```

Comment:
Hi @mmusgrov, do you think this setter is really needed? It isn't implemented elsewhere, so nothing will happen if called. :-)

Reply:
I don't know since it is a public interface, perhaps it gets used in the quarkus datasources extension or some other framework that want to supply its own implementation of the config.

Reply:
Actually you are correct, the setter for the property is done via AgroalDataSourceConfigurationSupplier and the interface is exposing the configured value. I will remove it.

Reply:
Btw can you also add it to the Spring Boot Data source. Thanks 🙏

Reply:
Hi @mmusgrov, your assumption is correct. Every agroal property needs a spring.datasource.agroal prefix.

Reply:
Setting `runner.withPropertyValues("spring.datasource.agroal.recovery-enable=false", ...` had no effect so I'm expecting some other setting is required. Could you git fetch my PR and see if you figure out how to set the property please?

Reply:
I cannot see an update to AgroalDatasource in the Spring Boot project. This is needed to get the property in.

Reply:
Thanks very much Benjamin for spotting that I'd forgotten to add it to AgroalDataSource, too many things going on!